grails-app/services/org/pih/warehouse/data/ProductSupplierDataService.groovy

```diff
@@ -141,9 +141,27 @@ class ProductSupplierDataService {
     }
 
     def getOrCreateNew(Map params) {
-        def productSupplier = params.productSupplier ? ProductSupplier.get(params.productSupplier) : null
+        def productSupplier
+        if (params.productSupplier) {
+            productSupplier = params.productSupplier ? ProductSupplier.get(params.productSupplier) : null
+        } else {
+            productSupplier = ProductSupplier.createCriteria().list {
+                eq("product", params.product)
+                if (params.supplier && params.supplierCode && params.product) {
+                    eq("supplier", params.supplier)
+                    eq("supplierCode", params.supplierCode.replaceAll('[,-]',''))
+                    // TODO we need to find a way to replace dashes in attribues, solutions below are not working
+                    // sqlRestriction("REPLACE('{alias}.supplierCode','-','')");
+                    // def code = params.supplierCode.replaceAll('[,-]','')
+                    // sqlRestriction("replace(supplierCode, '-', '') = '%$code%'")
+                } else if (!params.supplierCode) {
+                    eq("manufacturer", params.manufacturer)
+                    eq("manufacturerCode", params.manufacturerCode.replaceAll('[,-]',''))
+                }
+            }
+        }
         if (productSupplier) {
-            return productSupplier
+            return productSupplier instanceof ArrayList ? productSupplier.first() : productSupplier
```

Comment:
We should be able to do this in the query using maxResults(1)

Reply:
I think, that after adding `maxResult` it still will be a list. Would  it be also helpful if you additionally change the `ProductSupplier.createCriteria().list` into `.get`? Can it work together with `maxResult`?